File: Baekjun/8.5/1545. 안티 팰린드롬/solution.swift
func solution(){
  var S = readLine()!.map { Character(String($0)) }
  
  let n = S.count
  var i:Int = n / 2 - 1, j:Int = n / 2 + (n % 2 == 1 ? 1 : 0)
  
  S.sort()
  
  while j < n && i > -1 {
      var idx = j
      if S[i] == S[idx] {
          while S[i] == S[idx] {
              idx += 1
              if idx == n {
                  print(-1)
                  return
              }
          }
          S.swapAt(j, idx)
      }
      i -= 1
      j += 1
  }
  
  print(String(S))
}
solution()

File: Baekjun/8.5/1545. 안티 팰린드롬/solution.py
import sys

input = sys.stdin.readline

s = list(input().strip())
s.sort()  # 정렬
n = len(s)
i, j = n // 2 - 1, n // 2 + int(n % 2)

ans = 0

while i > -1 and j < n:
    idx = j  # 팰린드롬이면 j와 바꿀 index
    if s[i] == s[j]:  # 팰린드롬인 부분이라면
        while s[i] == s[idx]:
            idx += 1  # idx 1씩 늘려가며 j와 바꿀 위치 찾기
            if idx == n:  # 없다면
                ans = -1  # 불가능
                break
        if ans == -1:
            break
    s[j], s[idx] = s[idx], s[j]  # swap
    i -= 1
    j += 1

if not ans:
    ans = "".join(s)

print(ans)
